core/controllers/editor.py

```diff
@@ -272,8 +273,8 @@ class ExplorationHandler(EditorHandler):
     def _get_exploration_data(self, exploration_id, version=None):
         """Returns a description of the given exploration."""
         try:
-            exploration = exp_services.get_exploration_by_id(
-                exploration_id, version=version)
+            exploration = exp_services.get_exp_with_draft_applied(
```

Comment:
I believe the backend is responsible for checking the version here, and informing the front-end if there is a conflict. Also you should add a test that covers this situation.